### ISSUE TITLE
Issue properly terminating Train Network function

### ISSUE THREAD

**Issue by nschapp** [DeepLabCut/DeepLabCut#863]
**Your Operating system and DeepLabCut version**
Windows 10, with the latest update of DeepLabCut and Anaconda (both only installed a week ago)

**Describe the problem**
My issue isn't a specific error with a code I am encountering, but rather, a problem running the "train network" function.  I have followed the pipeline up to that function but I am unable to get the train network to terminate without losing all of its data.  I can copy here what I am seeing in the command window at the bottom of this post.

Perhaps I haven't properly used the train network function in this program, but I have attempted to stop the program manually by killing the function or the command window, but it results in no network training data being visible in the relevant folder afterwards, indicated by the following error when I attempt to perform the evaluate network function: 
 raise FileNotFoundError("Snapshots not found! It seems the dataset for shuffle %s and trainFraction %s is not trained.\nPlease train it before evaluating.\nUse the function 'train_network' to do so."%(shuffle,trainFraction))

If there is a need to specify as argument to limit or save the data at a certain point in the train network function, I must have missed it.  I would appreciate if this was just a user error on my part what I can change to run the function properly.

Command prompt output below:

In [7]: deeplabcut.train_network(path_config_file)
Config:
{'all_joints': [[0], [1], [2], [3], [4]],
 'all_joints_names': ['snout', 'right ear', 'left ear', 'body center', 'tail'],
 'batch_size': 1,
 'bottomheight': 400,
 'crop': True,
 'crop_pad': 0,
 'cropratio': 0.4,
 'dataset': 'training-datasets\\iteration-0\\UnaugmentedDataSet_ReachingAug17\\Reaching_Mackenzie95shuffle1.mat',
 'dataset_type': 'default',
 'deterministic': False,
 'display_iters': 1000,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': 'C:\\Users\\malavikalabuser\\Anaconda3\\envs\\DLC-CPU\\lib\\site-packages\\deeplabcut\\pose_estimation_tensorflow\\models\\pretrained\\resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'leftwidth': 400,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 0.05,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'max_input_size': 1500,
 'mean_pixel': [123.68, 116.779, 103.939],
 'metadataset': 'training-datasets\\iteration-0\\UnaugmentedDataSet_ReachingAug17\\Documentation_data-Reaching_95shuffle1.pickle',
 'min_input_size': 64,
 'minsize': 100,
 'mirror': False,
 'multi_step': [[0.005, 10000],
                [0.02, 430000],
                [0.002, 730000],
                [0.001, 1030000]],
 'net_type': 'resnet_50',
 'num_joints': 5,
 'optimizer': 'sgd',
 'pos_dist_thresh': 17,
 'project_path': 'C:\\Users\\malavikalabuser\\Reaching-Mackenzie-2020-08-17',
 'regularize': False,
 'rightwidth': 400,
 'save_iters': 50000,
 'scale_jitter_lo': 0.5,
 'scale_jitter_up': 1.25,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': 'C:\\Users\\malavikalabuser\\Reaching-Mackenzie-2020-08-17\\dlc-models\\iteration-0\\ReachingAug17-trainset95shuffle1\\train\\snapshot',
 'stride': 8.0,
 'topheight': 400,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
Switching batchsize to 1, as default/tensorpack/deterministic loaders do not support batches >1. Use imgaug loader.
Starting with standard pose-dataset loader.
Initializing ResNet
WARNING:tensorflow:From C:\Users\malavikalabuser\Anaconda3\envs\DLC-CPU\lib\site-packages\tensorflow\python\framework\op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From C:\Users\malavikalabuser\Anaconda3\envs\DLC-CPU\lib\site-packages\tensorflow\python\framework\op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From C:\Users\malavikalabuser\Anaconda3\envs\DLC-CPU\lib\site-packages\tensorflow\python\ops\losses\losses_impl.py:209: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From C:\Users\malavikalabuser\Anaconda3\envs\DLC-CPU\lib\site-packages\tensorflow\python\ops\losses\losses_impl.py:209: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
Loading ImageNet-pretrained resnet_50
WARNING:tensorflow:From C:\Users\malavikalabuser\Anaconda3\envs\DLC-CPU\lib\site-packages\tensorflow\python\training\saver.py:1266: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.
WARNING:tensorflow:From C:\Users\malavikalabuser\Anaconda3\envs\DLC-CPU\lib\site-packages\tensorflow\python\training\saver.py:1266: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.
INFO:tensorflow:Restoring parameters from C:\Users\malavikalabuser\Anaconda3\envs\DLC-CPU\lib\site-packages\deeplabcut\pose_estimation_tensorflow\models\pretrained\resnet_v1_50.ckpt
INFO:tensorflow:Restoring parameters from C:\Users\malavikalabuser\Anaconda3\envs\DLC-CPU\lib\site-packages\deeplabcut\pose_estimation_tensorflow\models\pretrained\resnet_v1_50.ckpt
Training parameter:
{'stride': 8.0, 'weigh_part_predictions': False, 'weigh_negatives': False, 'fg_fraction': 0.25, 'weigh_only_present_joints': False, 'mean_pixel': [123.68, 116.779, 103.939], 'shuffle': True, 'snapshot_prefix': 'C:\\Users\\malavikalabuser\\Reaching-Mackenzie-2020-08-17\\dlc-models\\iteration-0\\ReachingAug17-trainset95shuffle1\\train\\snapshot', 'log_dir': 'log', 'global_scale': 0.8, 'location_refinement': True, 'locref_stdev': 7.2801, 'locref_loss_weight': 0.05, 'locref_huber_loss': True, 'optimizer': 'sgd', 'intermediate_supervision': False, 'intermediate_supervision_layer': 12, 'regularize': False, 'weight_decay': 0.0001, 'mirror': False, 'crop_pad': 0, 'scoremap_dir': 'test', 'batch_size': 1, 'dataset_type': 'default', 'deterministic': False, 'crop': True, 'cropratio': 0.4, 'minsize': 100, 'leftwidth': 400, 'rightwidth': 400, 'topheight': 400, 'bottomheight': 400, 'all_joints': [[0], [1], [2], [3], [4]], 'all_joints_names': ['snout', 'right ear', 'left ear', 'body center', 'tail'], 'dataset': 'training-datasets\\iteration-0\\UnaugmentedDataSet_ReachingAug17\\Reaching_Mackenzie95shuffle1.mat', 'init_weights': 'C:\\Users\\malavikalabuser\\Anaconda3\\envs\\DLC-CPU\\lib\\site-packages\\deeplabcut\\pose_estimation_tensorflow\\models\\pretrained\\resnet_v1_50.ckpt', 'net_type': 'resnet_50', 'num_joints': 5, 'display_iters': 1000, 'max_input_size': 1500, 'metadataset': 'training-datasets\\iteration-0\\UnaugmentedDataSet_ReachingAug17\\Documentation_data-Reaching_95shuffle1.pickle', 'min_input_size': 64, 'multi_step': [[0.005, 10000], [0.02, 430000], [0.002, 730000], [0.001, 1030000]], 'pos_dist_thresh': 17, 'project_path': 'C:\\Users\\malavikalabuser\\Reaching-Mackenzie-2020-08-17', 'save_iters': 50000, 'scale_jitter_lo': 0.5, 'scale_jitter_up': 1.25, 'output_stride': 16, 'deconvolutionstride': 2}
Starting training....
iteration: 1000 loss: 0.0208 lr: 0.005
iteration: 2000 loss: 0.0099 lr: 0.005
iteration: 3000 loss: 0.0072 lr: 0.005
iteration: 4000 loss: 0.0055 lr: 0.005
iteration: 5000 loss: 0.0051 lr: 0.005
iteration: 6000 loss: 0.0043 lr: 0.005
iteration: 7000 loss: 0.0038 lr: 0.005
iteration: 8000 loss: 0.0036 lr: 0.005
iteration: 9000 loss: 0.0033 lr: 0.005
iteration: 10000 loss: 0.0032 lr: 0.005
iteration: 11000 loss: 0.0055 lr: 0.02
iteration: 12000 loss: 0.0041 lr: 0.02
iteration: 13000 loss: 0.0034 lr: 0.02
iteration: 14000 loss: 0.0030 lr: 0.02
iteration: 15000 loss: 0.0028 lr: 0.02
iteration: 16000 loss: 0.0026 lr: 0.02
iteration: 17000 loss: 0.0024 lr: 0.02
iteration: 18000 loss: 0.0024 lr: 0.02
iteration: 19000 loss: 0.0024 lr: 0.02
iteration: 20000 loss: 0.0023 lr: 0.02
iteration: 21000 loss: 0.0021 lr: 0.02
iteration: 22000 loss: 0.0021 lr: 0.02
iteration: 23000 loss: 0.0020 lr: 0.02
iteration: 24000 loss: 0.0020 lr: 0.02
iteration: 25000 loss: 0.0020 lr: 0.02
iteration: 26000 loss: 0.0019 lr: 0.02
iteration: 27000 loss: 0.0018 lr: 0.02
iteration: 28000 loss: 0.0019 lr: 0.02
iteration: 29000 loss: 0.0018 lr: 0.02
iteration: 30000 loss: 0.0018 lr: 0.02
iteration: 31000 loss: 0.0018 lr: 0.02
iteration: 32000 loss: 0.0018 lr: 0.02
iteration: 33000 loss: 0.0018 lr: 0.02
iteration: 34000 loss: 0.0017 lr: 0.02
iteration: 35000 loss: 0.0017 lr: 0.02
iteration: 36000 loss: 0.0017 lr: 0.02
iteration: 37000 loss: 0.0016 lr: 0.02
iteration: 38000 loss: 0.0016 lr: 0.02
iteration: 39000 loss: 0.0016 lr: 0.02
iteration: 40000 loss: 0.0016 lr: 0.02
iteration: 41000 loss: 0.0016 lr: 0.02
iteration: 42000 loss: 0.0016 lr: 0.02
iteration: 43000 loss: 0.0016 lr: 0.02
iteration: 44000 loss: 0.0015 lr: 0.02
iteration: 45000 loss: 0.0015 lr: 0.02
iteration: 46000 loss: 0.0015 lr: 0.02
iteration: 47000 loss: 0.0016 lr: 0.02
iteration: 48000 loss: 0.0015 lr: 0.02
iteration: 49000 loss: 0.0015 lr: 0.02
iteration: 50000 loss: 0.0014 lr: 0.02
//it continues onward from here if not interrupted//

**Comment by jeylau** [comment]
Hi @nschapp, you can pass `save_iters` to `train_network` to dictate how often snapshots should be saved. This effectively overwrites the same parameter in the pose_cfg.yaml you can find under the dlc-models/.../train directory. It is 50000 by default, so it seems you may have interrupted training just too early. Set it to a lower value, or let it train for 1000 more iterations (to ensure weights have time to get dumped) and you will be fine.
You could also verify you have write permissions on your machine, but I do not think this is the issue here 😉
